### ISSUE TITLE
Help Choosing Parameters for Custom Dataset

### ISSUE THREAD

**Issue by hockeybro12** [matterport/Mask_RCNN#1648]
I have trained a MaskRCNN and it's doing well, but it's not able to identify all the objects in my dataset. I have 20K images, but it's not able to overfit. Thus, I am looking for help choosing parameters, as I do have objects of various sizes.

The image size is 100 x 400. Here are my current parameters:
```
class Config(Config):

   name = "config

    NUM_CLASSES = 9 + 10

    GPU_COUNT = 1
    IMAGES_PER_GPU = 10

    # Use small images for faster training. Set the limits of the small side
    # the large side, and that determines the image shape.
    IMAGE_MIN_DIM = 64
    IMAGE_MAX_DIM = 256

    PRE_NMS_LIMIT = 4000

    IMAGE_RESIZE_MODE = 'none'

    RPN_TRAIN_ANCHORS_PER_IMAGE = 512

    # use smaller anchors as the objects are small
    RPN_ANCHOR_SCALES = (16, 32, 64, 128, 256)

    # scale up the image so that we can identify the smaller objects in the images better
    IMAGE_MIN_SCALE = 2.0
```

Since the performance isn't perfect, I ran the tutorial in the `inspect_data.ipynb` notebook. I ran the cell to show me the Anchors, and got this result:

![Screen Shot 2019-07-19 at 5 44 35 PM](https://user-images.githubusercontent.com/5779609/61572046-53676700-aa4d-11e9-9368-5356a2217497.png)

To me, it seems like there aren't enough anchors here, compared to the demo in the notebook. However, no matter how I change my anchor scales parameters, it doesn't change.

Does anyone have an idea of if my parameters are good or I should change them?

Thanks

**Comment by hockeybro12** [comment]
I fixed it. Needed `RPN_ANCHOR_SCALES = (4, 8, 16, 32, 64)` and `TRAIN_ROIS_PER_IMAGE = 32`.

**Comment by May-forever** [comment]
Dear hockeybro12,

I want to use Mask-RCNN to train on my custom dataset (just one class).

First of all, I use Labelme to create groundtruth for each image. Each image has a .json file as its ground-truth.

After that, I don't know what should I do...

I want to know how to convert my groundtruth file to a format that Mask-rcnn can use. Do you have any tools ?

Looking forward to hearing from you, thanks in advance.

With best regards,
Ming

**Comment by May-forever** [comment]
Dear hockeybro12,

I want to use Mask-RCNN to train on my custom dataset (just one class).

First of all, I use Labelme to create groundtruth for each image. Each image has a .json file as its ground-truth.

After that, I don't know what should I do...

I want to know how to convert my groundtruth file to a format that Mask-rcnn can use. Do you have any tools ?

Looking forward to hearing from you, thanks in advance.

With best regards,
Ming

**Comment by hockeybro12** [comment]
@May-forever You can look at this example to see how to create the mask for your dataset. You just need to subclass utils.Dataset and update those functions just like in the example with ShapesDataset.

https://github.com/matterport/Mask_RCNN/blob/master/samples/shapes/train_shapes.ipynb

**Comment by May-forever** [comment]
> @May-forever You can look at this example to see how to create the mask for your dataset. You just need to subclass utils.Dataset and update those functions just like in the example with ShapesDataset.
> 
> https://github.com/matterport/Mask_RCNN/blob/master/samples/shapes/train_shapes.ipynb

Thank you very much
